### PR TITLE
Add dynamic select for service name and entity [sc-163505]

### PR DESCRIPTION
**Pull request by alexbourret**
Story details: https://app.shortcut.com/dataiku/story/163505

### CODE REVIEW

**Review by Alexlandeau — CHANGES_REQUESTED** [review]
Plugin works fine, a few nitpicks / open questions on syntax though

File: python-lib/odata_client.py [hunk 1]
```diff
@@ -86,6 +85,7 @@ def get_session(self, config, odata_version):
         return session
 
     def get_entity_collections(self, entity, top=None, skip=None, page_url=None, filter=None):
+        entity = entity or ""
```
[the remark this answers is not in the document]
Comment (alexbourret) [reply python-lib/odata_client.py]: Fixed in [363fe9a](https://github.com/dataiku/dss-plugin-sap-odata/pull/11/commits/363fe9a3a6b0a4fd0bb74f0574c09cbbe5cf00b2)

File: python-lib/odata_common.py [hunk 2]
```diff
@@ -51,3 +53,74 @@ def get_clean_row_method(config):
...
+    if odata_service_node != "":
+        odata_instance = "/".join([login.get(ODataConstants.INSTANCE, "").strip("/"), odata_service_node])
+    else:
+        odata_instance = login.get(ODataConstants.INSTANCE, "").strip("/")
+    return odata_instance
+
+
+def get_list_title(config):
+    odata_list_title = None
+    odata_list_title = config.get("odata_list_selector")
```
[the remark this answers is not in the document]
Comment (alexbourret) [reply python-lib/odata_common.py]: Fixed in [363fe9a](https://github.com/dataiku/dss-plugin-sap-odata/pull/11/commits/363fe9a3a6b0a4fd0bb74f0574c09cbbe5cf00b2)

File: python-lib/odata_common.py [hunk 3]
```diff
@@ -51,3 +53,74 @@ def get_clean_row_method(config):
...
+    def _build_select_choices(self, choices=None):
+        if not choices:
+            return {"choices": []}
+        if isinstance(choices, str):
+            return {"choices": [{"label": "{}".format(choices)}]}
+        if isinstance(choices, list):
+            return {"choices": choices}
+        if isinstance(choices, dict):
+            returned_choices = []
+            for choice_key in choices:
```
[the remark this answers is not in the document]
Comment (alexbourret) [reply python-lib/odata_common.py]: Fixed in [363fe9a](https://github.com/dataiku/dss-plugin-sap-odata/pull/11/commits/363fe9a3a6b0a4fd0bb74f0574c09cbbe5cf00b2)

**Review by Alexlandeau — APPROVED** [review]
LGTM !

**Pull request merged**